modules/pool/lib/apr-data-sources/ib-linear-apr-handlers/sources/reaper-crypt-apr-handler.ts

```diff
@@ -0,0 +1,141 @@
+    }
+
+    async getAprs(): Promise<{ [p: string]: number }> {
+        let multiStrategyAprs = {};
+        let singleStrategyAprs = {};
+        this.wstETHBaseApr = await this.getWstEthBaseApr();
+        if (this.multiStrategyTokens !== undefined) {
+            multiStrategyAprs = await this.getMultiStrategyAprFromSubgraph(this.multiStrategyTokens);
+        }
+        if (this.singleStrategyTokens !== undefined) {
+            singleStrategyAprs = await this.getSingleStrategyCryptApr(this.singleStrategyTokens);
+        }
+        return { ...multiStrategyAprs, ...singleStrategyAprs };
+    }
+
+    private async getSingleStrategyCryptApr(tokens: {
+        [tokenName: string]: { address: string; isSftmX?: boolean; isWstETH?: boolean };
+    }): Promise<{ [tokenAddress: string]: number }> {
+        const aprs: { [tokenAddress: string]: number } = {};
+        for (const { address, isSftmX, isWstETH } of Object.values(tokens)) {
+            const tokenContract = getContractAt(address, ReaperCryptAbi);
+            const strategyAddress = await tokenContract.strategy();
+            const strategyContract = getContractAt(strategyAddress, ReaperCryptStrategyAbi);
+            let avgAprAcrossXHarvests = 0;
+
+            avgAprAcrossXHarvests =
+                (await strategyContract.averageAPRAcrossLastNHarvests(this.averageAPRAcrossLastNHarvests)) /
+                APR_PERCENT_DIVISOR;
+            if (isSftmX) {
+                avgAprAcrossXHarvests = avgAprAcrossXHarvests * (1 + sFTMxBaseApr);
+            }
+            if (isWstETH) {
+                avgAprAcrossXHarvests = avgAprAcrossXHarvests * (1 + this.wstETHBaseApr);
+            }
+            aprs[address] = avgAprAcrossXHarvests;
+        }
+
+        return aprs;
+    }
+
+    private async getMultiStrategyAprFromSubgraph(tokens: {
+        [tokenName: string]: { address: string; isSftmX?: boolean; isWstETH?: boolean };
+    }): Promise<{ [tokenAddress: string]: number }> {
+        const requestQuery = {
+            operationName: 'getVaults',
+            query: this.query,
+            variables: {
+                ids: Object.values(tokens).map(({ address }) => address),
+            },
+        };
+        const {
+            data: { data },
+        }: { data: { data: MultiStratResponse } } = await axios({
+            method: 'post',
+            url: this.subgraphUrl,
+            data: JSON.stringify(requestQuery),
+        });
+        return data.vaults.reduce((acc, { id, apr }) => {
+            const token = Object.values(tokens).find((token) => token.address.toLowerCase() === id.toLowerCase());
+            if (!token) {
+                return acc;
+            }
+            let tokenApr = parseFloat(apr) / APR_PERCENT_DIVISOR;
+            if (token.isSftmX) {
+                tokenApr = tokenApr * (1 + sFTMxBaseApr);
+            }
+            if (token.isWstETH) {
+                tokenApr = tokenApr * (1 + this.wstETHBaseApr);
+            }
```

Comment:
is there a possibility to make this more general? Like if any of the IB-token-apr handlers for linear pools have an IB token, this needs to apply.